### PR TITLE
chore: Drop the "Fixes" check in CI

### PR DESCRIPTION
**Pull request by vicb**
## What this PR solves / how to test

I had the PR description check failed because I typed "Fixes n/a" instead of "Fixes N/A"
It should be ok to make this check case insensitive.

Fixes N/A

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Tests included
  - [ ] Tests not necessary because: trivial change
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [X] Not required because: CI change
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Changeset included
  - [X] Changeset not necessary because: CI minor tweak
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [X] Documentation not necessary because: CI

### CODE REVIEW

File: tools/deployments/validate-pr-description.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function validateDescription(
 	}
 
 	if (
-		!/^Fixes (#\d+|N\/A|\[[A-Z]+-\d+\]\(https:\/\/jira\.cfdata\.org\/browse\/[A-Z]+-\d+\))/m.test(
+		!/^Fixes (#\d+|N\/A|\[[A-Z]+-\d+\]\(https:\/\/jira\.cfdata\.org\/browse\/[A-Z]+-\d+\))/im.test(
```
Comment (vicb) [inline tools/deployments/validate-pr-description.ts:35]: Should we also drop anchoring?

**Comment by emily-shen** [comment]
While we're here, should we just remove this check entirely? I'm not sure what the benefit to having it is. 

We can leave `Fixes #etc.`  in the template to nudge people into including it, but the check is just a bit annoying 😅 

cc @penalosa

**Comment by penalosa** [comment]
Yeah, that makes sense—we're ending up with a lot of Fixes N/A in PRs, which isn't all that useful. Let's remove this entirely.

**Comment by vicb** [comment]
Thanks for the feedback, I'll update the PR

**Review by penalosa — APPROVED** [review]

**Pull request merged**